### DISCUSSION TITLE
World Info Search Depth should support values past 1024

### DISCUSSION THREAD

**Discussion by Reithan** [LostRuins/lite.koboldai.net#75] [Ideas]
As title. It looks like most models being hosted on horde now support context up to 8192 and at least 512 output. With only 1024 depth for World Info triggers that could cover as little as 1 prompt + 1 reply, which isn't much in a long-running chat or story.

It would be nice to have this expanded to include a few higher values, maybe 2048 and 4096?

**Comment by LostRuins** [comment]
yeah i can add that

**Comment by LostRuins** [comment]
Alright, updated, please try

**Comment by Reithan** [comment]
Testing it now, it's not broken atm, but will need to test longer to see if works as intended. Just a quick question, went over the script real fast but does https://github.com/LostRuins/lite.koboldai.net/blob/main/index.html#L10640 return the cutoff context by token cutoff, or characters?

If it's characters, I might actually want some much high numbers, but I'm under the assumption based on prior testing that it is measure in tokens.

**Comment by Reithan** [comment]
I have a token that's about ~2500 tokens up the history at the very start of the current scene in my chat, I have the context depth set to 4096, but it's not triggering.
I'll try placing it further down and see if it triggers. If the context depth is actually characters, rather than tokens, it would be awesome to have some substantially higher values. 

As an example, this token that's 2500 characters up, is roughly 9700 characters up.

So, if this is characters, maybe adding options for 8092 and 16184 would be nice?

I tried running "Full Context" before, but it caught too much, 1024 catches far too little, I'd love something close to at least the halfway point. Even with 4096 it feels like I'd need to constantly have certain phrases repeat just to keep them in memory for the current scene.

BTW, basing token & character counts on https://belladoreai.github.io/llama-tokenizer-js/example-demo/build/

**Comment by Reithan** [comment]
For additional context the current chat story is roughly 28k tokens long and over 100k characters.
With "Full" context, it'll basically trigger every entry, which is obviously unusable.

**Comment by Reithan** [comment]
Ok, read through the rest of that section of code. Looks like everything done by character with an estimate of 3 chars = 1 token since no tokenizer is implemented.

However, for WI depth, it's just 1:1, rather than 1:3 like the other sections, and wi depth truncation is done before `truncated_context` is actually truncated here: https://github.com/LostRuins/lite.koboldai.net/blob/main/index.html#L10758

So, perhaps we could just multiply the WI Depth by 3 to estimate token length as we're doing elsewhere? I think most of the logs I've looked at the char to token conversion has been closer to 1:4, but that's a different discussion to rabbit hole on.

We may want to consider "full depth" for WI Depth to be the full submitted context, rather than the full TOTAL history, or make those 2 separate options on their own, as well.

**Comment by Reithan** [comment]
Tested a bit more by manually altering my save file.
16384 seems to be pretty good for my use, but 32768 was way too much

If we map those the 1:3 token conversion, that would be ~5461 and ~10922

SO, if you just add the 1:3 conversion to the code, that would be spot on for me, at least, and if you added a 8192 option above 4096 along with that, it should basically catch anything anyone wanted to do(realistically) I think?

**Comment by LostRuins** [comment]
Ah yes, all depth counts are in characters, same for the author note depth. I'll try to make that clearer.

I can probably add a 8192 option too
